components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/bean/User.java

```diff
@@ -308,7 +308,7 @@ public UserBuilder setClaimManager(ClaimManager claimManager) {
 
         public User build() {
 
-            if (userId == null || domain == null || tenantDomain == null || identityStore == null ||
+            if (userId == null || tenantDomain == null || identityStore == null ||
```

Comment:
User must have a domain. When authenticating the user, check for which domain user belongs, if not set the default domain.